### PR TITLE
net: coap: Fixed discovery response formatting according to RFC6690

### LINKED ISSUE #31609: CoAP discovery response does not follow CoRE link format specification
**Describe the bug**
CoAP RFC ([RFC7252](https://tools.ietf.org/html/rfc7252)) states that end points should support the CoRE Link Format of discoverable resources as described in [RFC6690](https://tools.ietf.org/html/rfc6690) (refer [section 7.2](https://tools.ietf.org/html/rfc7252#section-7.2) of the RFC7252).
However, the current CoAP implementation does not place comma “,” and semicolon “;” in appropriate places in the payload according to the RFC6690. 

**To Reproduce**
Follow the usual building and flashing steps for "samples/net/sockets/coap_server" application on any board.
The payload of the discovery response is as shown below.

`</test>;</seg1/seg2/seg3>;</wild1/+/wild3>;</wild2/#>;</query>;</separate>;</location-query>;</large>;</large-update>;</large-create>;</obs>;</core1>title="Core 1";rt=core1;</core2>title="Core 1";rt=core1;`

**Expected behavior**
According to the RFC6690, the payload should be formatted as follows.

`</test>,</seg1/seg2/seg3>,</wild1/+/wild3>,</wild2/#>,</query>,</separate>,</location-query>,</large>,</large-update>,</large-create>,</obs>,</core1>;title="Core 1";rt=core1,</core2>;title="Core 1";rt=core1`

**Impact**
CoAP client libraries which follow RFC6690 may fail to parse the discovery response.

### PR DESCRIPTION
**Pull request by kasunch**
This PR resolves #31609 

Signed-off-by: Kasun Hewage <kasun.ch@gmail.com>

### CODE REVIEW

File: subsys/net/lib/coap/coap_link_format.c [hunk 1]
```diff
@@ -535,6 +519,13 @@ int coap_well_known_core_get(struct coap_resource *resource,
 		if (r < 0) {
 			goto end;
 		}
+
+		if ((resource + 1) && (resource + 1)->path) {
+			r = append_to_coap_pkt(response, ",", 1, &remaining, &offset, ctx.current);
```
Comment (rlubos) [inline subsys/net/lib/coap/coap_link_format.c:524]: nit, this line is a bit lengthy, let's break it to fit into char limit:
```suggestion
			r = append_to_coap_pkt(response, ",", 1, &remaining, 
					       &offset, ctx.current);
```

**Review by rlubos — COMMENTED** [review]
Looks good, just one nit. Thanks for addressing this issue!

File: subsys/net/lib/coap/coap_link_format.c [hunk 2]
```diff
@@ -347,36 +346,21 @@ static int format_attributes(const char * const *attributes,
 			return 0;
 		}
 
-		attr++;
-		if (!*attr) {
-			continue;
-		}
+		int attr_len = strlen(*attr);
```
Comment (jukkar) [inline subsys/net/lib/coap/coap_link_format.c:349]: Please introduce variables at the beginning of the block

**Comment by jukkar** [comment]
Please, no merge commits in PRs. Just rebase your stuff against latest master and force push to this PR.

File: subsys/net/lib/coap/coap_link_format.c (hunk 1 above)
Comment (kasunch) [reply subsys/net/lib/coap/coap_link_format.c]: Done Thanks

**Comment by kasunch** [comment]
I will rebase and push.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by rlubos — APPROVED** [review]

**Comment by kasunch** [comment]
Lookd like only the build 14 was failed. Any idea?

**Pull request merged**